app/routes/users/components/PhotoConsents.js

```diff
@@ -0,0 +1,182 @@
+// @flow
+
+import { useState } from 'react';
+import Select from 'react-select';
+import Button from 'app/components/Button';
+import Flex from 'app/components/Layout/Flex';
+import { ConfirmModalWithParent } from 'app/components/Modal/ConfirmModal';
+import type { PhotoConsent, PhotoConsentDomain } from 'app/models';
+import moment from 'moment-timezone';
+import cx from 'classnames';
+import {
+  PHOTO_CONSENT_DOMAINS,
+  toReadableSemester,
+} from 'app/routes/events/utils';
+import styles from './PhotoConsents.css';
+
+const ConsentManager = ({
+  consent,
+  updateConsent,
+  isMe,
+}: {
+  consent: ?PhotoConsent,
+  updateConsent: (consent: PhotoConsent) => Promise<*>,
+  isMe: boolean,
+}) => {
+  if (!consent) {
+    return null;
+  }
+
+  const getConsentStatus = () => {
+    if (consent.isConsenting) {
+      return (
+        <>
+          <b>Du ga samtykket den </b>
+          <i>{moment(consent.updatedAt).format('DD. MMM YYYY')}.</i>
+        </>
+      );
+    } else if (consent.isConsenting === false) {
+      return (
+        <>
+          <b>Du trakk samtykket den </b>
+          <i>{moment(consent.updatedAt).format('DD. MMM YYYY')}.</i>
+        </>
+      );
+    }
+    return <b>Du har ikke tatt stilling til samtykket.</b>;
+  };
```

Comment:
why is this a function? It should just define a variable with the correct contents if you absolutely have to do it this way. And it will update on prop updates

Reply:
Yeah, I simplified this quite a bit. Should be a bit cleaner now. 😄